### PR TITLE
Change labels for the time periods

### PR DESCRIPTION
**Pull request by madsnedergaard**
## Issue

This has been a recurring issue, where the current usage of (24 hours, 30 days, etc.) is easily misunderstood and unclear. Using "hourly", "daily", etc. labels is also not perfect, but after internal discussions we've decided to try it out and see if it works better :)

## Description

Changes labels to "hourly", "daily", "monthly" and "yearly".
I added machine-translations for the most used languages, but would be great with any feedback from native speakers :)

**TODO:**

- [x] Update graph labels
- [x] Remove old inaccurate translations
- [ ] Update time slider title
- [ ] Update screenshot in onboarding modal
- [ ] Update text in onboarding modal

### Preview

Screenshot 2023-12-07 at 11 36 16

### CODE REVIEW

File: web/public/locales/es.json
Comment (silimotion) [inline web/public/locales/es.json:79]:
```suggestion
    "hourly": "horario",
```
I think this is better

Comment (wobniarin) [reply web/public/locales/es.json:79]: I agree with @silimotion :) In Spanish we say "diario" and "horario"

**Comment by VIKTORVAV99** [comment]
> Another wording option: "by hour"/"by day"/"by month"/"by year". ('per hour' etc. was yet another suggestion by DeepL.)
> 
> ChatGPT suggests 'by hour' (etc.) might be the most appropriate:
> 
> > The choice between using terms like **"hourly," "daily," "monthly," "yearly"** and **"per hour," "per day," "per month,"** as well as **"by hour," "by day," "by month,"** depends on the context and the specific information you want to convey. Here's a breakdown:
> > 
> > 1. **Hourly, Daily, Monthly, Yearly:**
> >    
> >    * **Hourly:** Refers to something that occurs every hour.
> >    * **Daily:** Refers to something that occurs every day.
> >    * **Monthly:** Refers to something that occurs every month.
> >    * **Yearly:** Refers to something that occurs every year.
> >    
> >    These terms are often used when describing frequency or rates.
> > 2. **Per Hour, Per Day, Per Month, Per Year:**
> >    
> >    * **Per Hour:** Indicates a rate or quantity for each hour.
> >    * **Per Day:** Indicates a rate or quantity for each day.
> >    * **Per Month:** Indicates a rate or quantity for each month.
> >    * **Per Year:** Indicates a rate or quantity for each year.
> >    
> >    These terms are used to express a rate or cost over a specific time period.
> > 3. **By Hour, By Day, By Month, By Year:**
> >    
> >    * **By Hour:** Indicates something is measured or calculated on an hourly basis.
> >    * **By Day:** Indicates something is measured or calculated on a daily basis.
> >    * **By Month:** Indicates something is measured or calculated on a monthly basis.
> >    * **By Year:** Indicates something is measured or calculated on a yearly basis.
> >    
> >    "By" in this context implies measurement or calculation according to the specified time unit.
> > 
> > Choose the appropriate term based on the context and clarity of communication. For example:
> > 
> > * "I get paid hourly" implies you receive payment for each hour worked.
> > * "The rent is $1,000 per month" indicates a fixed amount due each month.
> > * "We measure energy consumption by day" means tracking energy usage on a daily basis.

This is a good point and something I think we should consider. But I do feel like ChatGPTs response is missing that when speaking about temporal resolution hourly, daily, weekly, monthly, yearly does make sense. 

I got this from ChatGPT:
> The choice of wording for representing temporal resolution depends on the context and your target audience. However, for prioritizing understandability and translatability, "hourly, daily, monthly, yearly" is a good option. It provides clear and concise terms that are commonly understood in English and are likely to be easily translatable into other languages.
>
> This option is straightforward and aligns with common language conventions for expressing time intervals. It also minimizes the chances of confusion, making it accessible to a broad audience.

File: web/public/locales/fr.json
Comment (lav-julien) [inline web/public/locales/fr.json:1858]:
```suggestion
    "daily": "par jour",
```

File: web/public/locales/fr.json
Comment (lav-julien) [inline web/public/locales/fr.json:1859]:
```suggestion
    "hourly": "par heure",
```

File: web/public/locales/fr.json
Comment (lav-julien) [inline web/public/locales/fr.json:1860]:
```suggestion
    "monthly": "par mois",
```

File: web/public/locales/fr.json
Comment (lav-julien) [inline web/public/locales/fr.json:1861]:
```suggestion
    "yearly": "par année"
```

File: web/public/locales/en.json
Comment (VIKTORVAV99) [inline web/public/locales/en.json:94]: I think we need to change this title a bit, it kind of clashes with the new wording.

Maybe "Select time resolution" or something along those lines?

**Review by VIKTORVAV99 — COMMENTED** [review]
Should we update this part in the chartTitle.ts file as well?
```jsx
      <h3 className="pt-4 text-md font-bold">
        {localExists
          ? __(`${translationKey}.${timeAverage}`)
          : __(
              `${translationKey}.default`,
              formatTimeRange(localDefaultExists ? i18n.language : 'en', timeAverage)
            )}
      </h3>
```

It also uses the current format/function.

**Comment by VIKTORVAV99** [comment]
@madsnedergaard shall we try and get this merged before the holiday code freeze or do you want to wait until after?

File: web/public/locales/fr.json
Comment (madsnedergaard) [reply web/public/locales/fr.json:1858]: Doesn't this mean "per hour" instead of "hourly"? I think we can do either way as @q-- mentioned, but I'd prefer it to be somewhat consistent across languages :)

File: web/public/locales/en.json
Comment (madsnedergaard) [reply web/public/locales/en.json:94]: Good point on changing it. I think "time resolution" is too technical for most people though :)

I'm also wondering if we should do this in a separate PR afterwards instead, as it could potentially be a longer discussion and requires new translations?

**Comment by madsnedergaard** [comment]
> @madsnedergaard shall we try and get this merged before the holiday code freeze or do you want to wait until after?

There's a lot of changes in recent contrib update, and I don't want to push them all to production just before going on vacation - so I'll merge this PR after :)

**Comment by VIKTORVAV99** [comment]
> > @madsnedergaard shall we try and get this merged before the holiday code freeze or do you want to wait until after?
> 
> 
> 
> There's a lot of changes in recent contrib update, and I don't want to push them all to production just before going on vacation - so I'll merge this PR after :)

I think those where already merged though 😅

Shouldn't be any big changes in control right now, just some parser upgrades by me and Robin.

File: web/public/locales/ru.json
Comment (svetakuzm) [inline web/public/locales/ru.json:1]: "daily": "по дням",
    "hourly": "по часам",
    "monthly": "по месяцам",
    "yearly": "по годам",

File: web/public/locales/ru.json
Comment (madsnedergaard) [inline web/public/locales/ru.json:43]:
```suggestion
    "daily": "по дням",
    "hourly": "по часам",
    "monthly": "по месяцам",
    "yearly": "по годам"
```

File: web/public/locales/ru.json
Comment (madsnedergaard) [reply web/public/locales/ru.json:1]: Brilliant, thanks @svetakuzm! 🙌

**Comment by madsnedergaard** [comment]
> Should we update this part in the chartTitle.ts file as well?
> 
> ```js
>       <h3 className="pt-4 text-md font-bold">
>         {localExists
>           ? __(`${translationKey}.${timeAverage}`)
>           : __(
>               `${translationKey}.default`,
>               formatTimeRange(localDefaultExists ? i18n.language : 'en', timeAverage)
>             )}
>       </h3>
> ```
> 
> It also uses the current format/function.

Yes we should, especially since many of them are quite outdated (saying 5 years instead of 7) 😅

**Comment by VIKTORVAV99** [comment]
@madsnedergaard if you don't have the time I'd be happy to pick up the last changes needed in this PR so we can deploy it in conjunction with the estimated labels.

**Comment by madsnedergaard** [comment]
> @madsnedergaard if you don't have the time I'd be happy to pick up the last changes needed in this PR so we can deploy it in conjunction with the estimated labels.

I would really appreciate that! :pray:
As you know a few other high priority things came in the way for me, so not sure when I would be able to pick this up again :)

**Comment by VIKTORVAV99** [comment]
> > @madsnedergaard if you don't have the time I'd be happy to pick up the last changes needed in this PR so we can deploy it in conjunction with the estimated labels.
> 
> I would really appreciate that! 🙏 As you know a few other high priority things came in the way for me, so not sure when I would be able to pick this up again :)

I know :)

I'll try and take care of it as soon as possible so we can get a mobile build out with all the new stuff in one go.

File: web/public/locales/en.json
Comment (VIKTORVAV99) [reply web/public/locales/en.json:149]: Hey @electricitymaps/advocacy what do you guys think here?

Should we try and come up with a new title in this PR or should we hold off and do it separately later on?

PS: Trying our new stuff 😅

**Review by VIKTORVAV99 — APPROVED** [review]
I'm gonna approve this since it works and then I'll let you decide when to merge it :)

But I am in favour of sooner rather than later and then clean up things in follow up PRs.

**Pull request merged**
